localstack/services/events/provider.py

```diff
@@ -596,12 +688,21 @@ def _get_limited_dict_and_next_token(
         return limited_dict, next_token
 
     def _extract_event_bus_name(
-        self, event_bus_name_or_arn: EventBusNameOrArn | None
+        self, resource_arn_or_name: EventBusNameOrArn | RuleArn | None
     ) -> EventBusName:
         """Return the event bus name. Input can be either an event bus name or ARN."""
-        if not event_bus_name_or_arn:
+        if not resource_arn_or_name:
+            return "default"
+        if "arn:aws:events" not in resource_arn_or_name:
+            return resource_arn_or_name
+        resource_type = get_bus_or_rule(resource_arn_or_name)
+        # TODO how to deal with / in event bus name or rule name
```

Comment:
This seems like a recurring theme (see https://github.com/localstack/localstack/pull/10789#discussion_r1600181172). Can we get that fixed before we introduce this bug at too many different places?

Reply:
Not really, I do not have any approach to fixing this, and we seam to also assume in many other providers that names will not contain /